internal/modules/oracle/test.py

```diff
@@ -121,6 +124,34 @@ def test_votes(self):
         for rate in vote_1["aggregate_vote"]["exchange_rate_tuples"]:
             self.assertEqual(float(rate["exchange_rate"]), float(EXCHANGE_RATES.GetRate(rate["denom"])))
 
+    # test_price_spread tests sending a large spread of prices and verifies
+    # the correct median price is chosen
+    def test_price_spread(self):
+        validators = [
+            {'home': validator1_home, 'name': validator1_val['name'], 'address': validator1_val['address']},
+            {'home': validator2_home, 'name': validator2_val['name'], 'address': validator2_val['address']},
+            {'home': validator3_home, 'name': validator3_val['name'], 'address': validator3_val['address']},
+        ]
+        exchange_rates = [
+            ExchangeRates(ExchangeRate("UMEE", "0.01"), ExchangeRate("ATOM", "1.00"), ExchangeRate("JUNO", "1.50")),
+            ExchangeRates(ExchangeRate("UMEE", "0.05"), ExchangeRate("ATOM", "50.00"), ExchangeRate("JUNO", "2.50")),
+            ExchangeRates(ExchangeRate("UMEE", "20.00"), ExchangeRate("ATOM", "3.00"), ExchangeRate("JUNO", "69.99")),
+        ]
+        wait_for_next_voting_period()
+        with concurrent.futures.ThreadPoolExecutor(3) as executor:
+            futures = []
+            for i in range(3):
+                futures.append(executor.submit(tx_send_prevote_and_vote, validators[i], exchange_rates[i]))
+            for future in concurrent.futures.as_completed(futures):
+                status, response = future.result()
+                self.assertTrue(status)
+        wait_for_next_voting_period(int(response['height']))
+        status, new_rates = query_exchange_rates()
+        self.assertTrue(status)
+        expected_rates = { 'ATOM': '3.0', 'JUNO': '2.5', 'UMEE': '0.05' }
```

Comment:
move this const to the top of the file please

Reply:
I like your other idea of auto-generating these :)